templates/scripts/setup-user-password.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+#
+# This script to help Admin initialize password for new users
+#
+
+function usage() {
+  echo "  Usage:"
+  echo "    $0 -g|--group <group> -u|--user <user> -h"
+  exit 1
+}
+
+function warning_exit() {
+  echo "WARNING: $1"
+  exit 2
+}
+
+function confirm() {
+  local _prompt _default _response
+ 
+  if [ "$1" ]; then _prompt="$1"; else _prompt="Are you sure"; fi
+  _prompt="$_prompt [y/n] ?"
+ 
+  # Loop forever until the user enters a valid response (Y/N or Yes/No).
+  while true; do
+    read -r -p "$_prompt " _response
+    case "$_response" in
+      [Yy][Ee][Ss]|[Yy]) # Yes or Y (case-insensitive).
+        return 0
+        ;;
+      [Nn][Oo]|[Nn])  # No or N.
+        return 1
+        ;;
+      *) # Anything else (including a blank) is invalid.
+        ;;
+    esac
+  done
+}
+
+# Parse args
+while getopts 'g:u:h' c
+do
+  case $c in
+    h) usage ;;
+    g) INPUT_GROUP=$OPTARG ;;
+    u) INPUT_USER=$OPTARG ;;
+  esac
+done
+
+[[ -z "$INPUT_GROUP" ]] && [[ -z "$INPUT_USER" ]] && INPUT_GROUP="console-allowed" # set default group
+
+[[ -n "$INPUT_GROUP" ]] && [[ -n "$INPUT_USER" ]] && warning_exit "Can not input both group and user togeher"
+
+# Collect users in this group without password
+function set_temp_password() {
+    user=$1
+    tpass=$(aws secretsmanager get-random-password --password-length 16 --require-each-included-type --output text)
+    aws iam get-login-profile --user-name $user >& /dev/null
+    if [ $? -eq 0 ]; then
```

Comment:
If the user already has a password please just skip, we don't want to wipe out a legitimate password. Actually could you just do this check earlier and remove them from the list if they already have a login profile?

Reply:
this makes sense... I thought to enforce this change if user wants it.

Reply:
Confirmed: the user list targets people whose .PasswordLastUsed == null, which means NoPasswordYet or Password expired, based on document explanation:
```aws iam list-users returns a list of users, each with a PasswordLastUsed value. If the value is missing, then the user either has no password or the password has not been used since IAM began tracking password age on October 20, 2014.```

So we don't wipe out a legitimate password.

Reply:
after more studies and testings, found there are others cases around profile (yes/no) vs password(yes/no). So I think we can just remove it from the list if already have a login-profile.